src/plugins/data/common/search/aggs/buckets/lib/time_buckets/calc_es_interval.ts

```diff
@@ -71,3 +72,90 @@ export function convertIntervalToEsInterval(interval: string): EsInterval {
     expression: interval,
   };
 }
+
+declare module 'moment' {
+  interface Locale {
+    _config: moment.LocaleSpecification;
+  }
+}
+
+// Below 5 seconds the "humanize" call returns the "few seconds" sentence, which is not ok for ms
+// This special config rewrite makes it sure to have precision also for sub-seconds durations
+// ref: https://github.com/moment/moment/issues/348
+export function getPreciseDurationDescription(
+  intervalValue: number,
+  unit: moment.unitOfTime.Base
+): string {
+  // moment cannot format anything below seconds, so this requires a manual handling
+  if (unit === 'millisecond') {
+    return intervalValue === 1
+      ? i18n.translate('data.search.aggs.buckets.intervalOptions.millisecond', {
+          defaultMessage: 'millisecond',
+        })
+      : i18n.translate('data.search.aggs.buckets.intervalOptions.milliseconds', {
+          defaultMessage: '{n} milliseconds',
+          values: { n: intervalValue },
+        });
```

Comment:
In case of a single unit, it should not return the actual number.
i.e.
```
timestamp per second => ok
timestamp per 1 second => not ok
```

Expressing that with the i18n syntax is not super easy to read/maintain.